src/net/driver/mod.rs

```diff
@@ -51,14 +55,14 @@ impl Reactor {
         };
 
         // Register a dummy I/O handle for waking up the polling thread.
-        let entry = reactor.register(&reactor.notify_reg.0)?;
+        let entry = reactor.register(&reactor.notify_reg.0, None)?;
         reactor.notify_token = entry.token;
 
         Ok(reactor)
     }
 
     /// Registers an I/O event source and returns its associated entry.
-    fn register(&self, source: &dyn Evented) -> io::Result<Arc<Entry>> {
+    pub fn register(&self, source: &dyn Evented, fd: Option<c_int>) -> io::Result<Arc<Entry>> {
```

Comment:
`Evented` shouldn't be part of our pub interface, and `fd` should be `RawFd`.

Reply:
This module is `pub(crate)`, and RawFd is Unix-only. I didn't want to copy/paste the method and struct, though perhaps that's preferable for memory usage? It'd be heavily preferable to use `Option<!>` instead of having to put every call in a cfg, though.